src/main/java/io/paradoxical/common/valuetypes/Value.java

```diff
@@ -0,0 +1,43 @@
+package io.paradoxical.common.valuetypes;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Base abstract class for wrappers on values allowing to give them their own
+ * type. This class and its subclasses are meant to be immutable by themselves -
+ * not allowing to change the reference to held value. This class cannot give
+ * any guarantees about the value itself.
+ *
+ * @author Tomasz Guzik <tomek@tguzik.com>
+ * @see com.tguzik.value.adapters.JaxbValueAdapter
+ * @since 0.1
+ */
+@SuppressWarnings("WeakerAccess")
```

Comment:
no, its because we don't really have good tests, so the only code referencing these code is in the project/libary, so it says it could be a package class, etc.